src/main/java/net/openhft/chronicle/threads/EventGroup.java

```diff
@@ -279,6 +279,13 @@ public void addTimingMonitor(final String description,
     protected void performStart() {
         if (core != null)
             core.start();
+
+        while (!isAlive()) {
+            Jvm.pause(1); // Busy-ish wait - not safe to allow other loops to start before core
+            if (core.isStopped())
+                throw new IllegalStateException("Core already stopped");
```

Comment:
I think if we throw in performStart we should be setting the lifecycle accordingly, probably needs to be done in AbstractLifecycleEventLoop e.g.
```
if (lifecycle.compareAndSet(EventLoopLifecycle.NEW, EventLoopLifecycle.STARTED)) {
    try {
        performStart();
    } catch (Exception e) {
        lifecycle.set(EventLoopLifecycle.STOPPED);
        Jvm.rethrow(e);
    }
}
```

Then you shouldn't need to `ignoreException("awaitTermination() interrupted");` below

Reply:
The expected lifecycle transitions are documented in `net.openhft.chronicle.threads.EventLoopLifecycle` and other code relies on that as an interface, hence the weird error when the contract is broken.

Reply:
I did that change but the test still produces that warning. I'm not sure why because every time I debug the code, the thread is not interrupted as implied.